applications/FSIapplication/custom_utilities/partitioned_fsi_utilities.hpp

```diff
@@ -120,81 +128,39 @@ class PartitionedFSIUtilities
-     * This function sets the variable data contained in a vector over the the
-     * fluid interface.
-     * @param rInterfaceModelPart: interface modelpart in where the vector variable is set
-     * @param rVariable: variable to be set
-     * @param rInterfaceDataVector: vector containing the data values to be set
+     * This function resizes and sets to zero an interface vector (length equal to the
+     * residual size).
+     * @param rInterfaceModelPart: interface modelpart in where the residual is computed
+     * @param pInterfaceVector: pointer to the vector that is to be resized
      */
-    void SetInterfaceVectorVariable(ModelPart& rInterfaceModelPart,
-                                    const Variable<array_1d<double, 3 > >& rVariable,
-                                    const VectorType& rInterfaceDataVector)
+    virtual void SetUpInterfaceVector(ModelPart& rInterfaceModelPart,
+                                      VectorPointerType& pInterfaceVector)
     {
-        // Initialize the variable value
-        VariableUtils().SetToZero_VectorVar(rVariable, rInterfaceModelPart.Nodes());
-
-        #pragma omp parallel for
-        for(int k=0; k<static_cast<int>(rInterfaceModelPart.NumberOfNodes()); ++k)
+        unsigned int ResidualSize = this->GetInterfaceResidualSize(rInterfaceModelPart);
+        if ( TSpace::Size(*pInterfaceVector) != ResidualSize )
         {
-            ModelPart::NodeIterator it_node = rInterfaceModelPart.NodesBegin()+k;
-            unsigned int base_i = k*TDim;
-
-            array_1d<double,3>& value_to_set = it_node->FastGetSolutionStepValue(rVariable);
-            for (unsigned int jj=0; jj<TDim; ++jj)
-            {
-                value_to_set[jj] = rInterfaceDataVector[base_i+jj];
-            }
+            TSpace::Resize(pInterfaceVector,ResidualSize);
         }
-    }
-
-    /**
-     * This function sets the variable data contained in a vector over the the
-     * fluid interface. The variable can be fixed or not.
-     * @param rInterfaceModelPart: interface modelpart in where the vector variable is set
-     * @param rVariable: variable to be set
-     * @param rFluidInterfaceDataVector: vector containing the data values to be set
-     * @param FixVariable: if true, fixes the variable in the fluid interface model part
-     */
-    void SetAndFixInterfaceVectorVariable(ModelPart& rInterfaceModelPart,
-                                          const Variable<array_1d<double, 3 > >& rVariable,
-                                          const VectorType& rFluidInterfaceDataVector,
-                                          const bool FixVariable)
-    {
-        this->SetInterfaceVectorVariable(rInterfaceModelPart, rVariable, rFluidInterfaceDataVector);
 
-        // If needed, apply fixity to rVariable
-        if (FixVariable)
-        {
-            // Get the variable components
-            typedef VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > > component_type;
```

Comment:
@pooyan-dadvand , we should find a good way of doing this. (@ruben please ignore this comment, is not for you)

Reply:
I agree, Actually I want to remove the components in the new data structure and use variable<double> instead with some additional information in the key.